File: app/src/main/java/com/xjl/eimui_demo/adapter/MainAdapter.java
package com.xjl.eimui_demo.adapter;

import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.xjl.eimui_demo.R;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class MainAdapter extends RecyclerView.Adapter<MainAdapter.ViewHolder> {

    List<String> list;
    Activity activity;
    LayoutInflater inflater;

    public MainAdapter(List<String> list, Activity activity) {
        this.list = list;
        this.activity = activity;
        inflater = this.activity.getLayoutInflater();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(inflater.inflate(R.layout.item_main, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.button.setText(list.get(position));
        holder.button.setOnClickListener(v -> {
            if (clickListener != null) {
                clickListener.onClicked(v, position);
            }
        });
    }

    private ClickListener clickListener;

    public void setClickListener(ClickListener clickListener) {
        this.clickListener = clickListener;
    }

    public interface ClickListener {
        void onClicked(View v, int position);
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        Button button;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            button = itemView.findViewById(R.id.button);
        }
    }

}

File: app/src/main/java/com/xjl/eimui/demo/adapter/MainAdapter.kt
package com.xjl.eimui.demo.adapter

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.recyclerview.widget.RecyclerView
import com.xjl.eimui.demo.R


class MainAdapter(var list: List<String>, activity: Activity) :
    RecyclerView.Adapter<MainAdapter.ViewHolder>() {

    var inflater: LayoutInflater
    init {
        inflater = activity.layoutInflater
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(inflater.inflate(R.layout.item_main, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.button.text = list[position]
        holder.button.setOnClickListener { v: View? ->
            if (clickListener != null) {
                clickListener!!.onClicked(v, position)
            }
        }
    }

    private var clickListener: ClickListener? = null


    fun setClickListener(clickListener: ClickListener?) {
        this.clickListener = clickListener
    }

    interface ClickListener {
        fun onClicked(v: View?, position: Int)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var button: Button
        init {
            button = itemView.findViewById(R.id.button)
        }
    }
}
